uni/lib/view/home/home.dart

```diff
@@ -15,4 +25,45 @@ class HomePageViewState extends GeneralPageViewState {
   Widget getBody(BuildContext context) {
     return MainCardsList();
   }
+
+  @override
+  Future<void> handleRefresh(BuildContext context) async {
+    final homePageProvider =
+        Provider.of<HomePageProvider>(context, listen: false);
+
+    final providersToUpdate = <StateProviderNotifier>{};
+
+    for (final cardType in homePageProvider.favoriteCards) {
+      switch (cardType) {
+        case FavoriteWidgetType.account:
+          providersToUpdate
+              .add(Provider.of<ProfileProvider>(context, listen: false));
+          providersToUpdate
+              .add(Provider.of<ReferenceProvider>(context, listen: false));
+          break;
+        case FavoriteWidgetType.exams:
+          providersToUpdate
+              .add(Provider.of<ExamProvider>(context, listen: false));
+          break;
+        case FavoriteWidgetType.schedule:
+          providersToUpdate
+              .add(Provider.of<LectureProvider>(context, listen: false));
+          break;
+        case FavoriteWidgetType.printBalance:
+          providersToUpdate
+              .add(Provider.of<ProfileProvider>(context, listen: false));
+          break;
+        case FavoriteWidgetType.libraryOccupation:
+          providersToUpdate.add(
+              Provider.of<LibraryOccupationProvider>(context, listen: false));
+          break;
+        case FavoriteWidgetType.busStops:
+          providersToUpdate
+              .add(Provider.of<BusStopProvider>(context, listen: false));
+          break;
+      }
```

Comment:
I don't like this logic very much :/ since this could be abstracted on the enum itself (at least in theory) but then you can't use Provider.of(). This makes future widgets on the homepage a bit of a hassle to add because this step can be overlooked easily.

Not sure if there's another way to handle this, that makes future features easier to add. If there aren't any possible ways to do it, I'll accept it

Reply:
I'm not a huge fan of this logic either. Do you think forcing an override of an hypothetical `refreshCard` in the Card widget would be more maintainable? I'm not seeing any other solution besides this.